packages/adminrouter/extra/src/test-harness/modules/mocker/endpoints/mesos_dns.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (C) Mesosphere, Inc. See LICENSE file for details.
+
+"""MesosDNS mock endpoint"""
+
+import copy
+import logging
+import re
+
+from exceptions import EndpointException
+from mocker.endpoints.recording import (
+    RecordingHTTPRequestHandler,
+    RecordingTcpIpEndpoint,
+)
+
+# pylint: disable=C0103
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=R0903
+class MesosDnsHTTPRequestHandler(RecordingHTTPRequestHandler):
+    """Request handler that mimics MesosDNS
+
+       Depending on how it was set up, it will respond with different SRV
+       entries for preset services.
+    """
+    SRV_QUERY_REGEXP = re.compile('^/v1/services/_([^_]+)._tcp.marathon.mesos$')
+
+    def _calculate_response(self, base_path, url_args, body_args=None):
+        """Reply with the currently set mock-reply for given SRV record query.
+
+        Please refer to the description of the BaseHTTPRequestHandler class
+        for details on the arguments and return value of this method.
+
+        Raises:
+            EndpointException: request URL path is unsupported
+        """
+
+        match = self.SRV_QUERY_REGEXP.search(base_path)
+        if match:
+            return self.__srv_permissions_request_handler(match.group(1))
+
+        raise EndpointException(
+            code=500,
+            content="Path `{}` is not supported yet".format(base_path))
```

Comment:
❓ It seems that there is little contention on the proposed style guide that we will use `{named_format_arguments}`, so maybe it would be good to start now.